app/models/mentor_training.rb

```diff
@@ -0,0 +1,31 @@
+# == Schema Information
+#
+# Table name: mentor_trainings
+#
+#  id              :uuid             not null, primary key
+#  date_completed  :datetime
+#  hours_completed :integer
+#  training_type   :enum
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  claim_id        :uuid
+#  provider_id     :uuid
+#  teacher_id      :uuid
+#
+# Indexes
+#
+#  index_mentor_trainings_on_claim_id     (claim_id)
+#  index_mentor_trainings_on_provider_id  (provider_id)
+#  index_mentor_trainings_on_teacher_id   (teacher_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (claim_id => claims.id)
+#  fk_rails_...  (provider_id => providers.id)
+#  fk_rails_...  (teacher_id => teachers.id)
+#
+class MentorTraining < ApplicationRecord
+  belongs_to :claim, class_name: "Claim"
+  belongs_to :teacher, optional: true
+  belongs_to :provider, optional: true
```

Comment:
Why are these two associations optional? 🤔

I'd have thought Mentor Training required a Mentor and a Provider, and that Claim was optional.

Reply:
It's because when creating a mentor_training in the provider step for example, we don't have a mentor on the mentor training for example. So there is a state where a mentor training has a provider or a mentor without the other.